### DISCUSSION TITLE
Welcome to SharpHook Discussions!

### DISCUSSION THREAD

**Discussion by TolikPylypchuk** [TolikPylypchuk/SharpHook#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by gulshan-verma** [comment]
Hi @TolikPylypchuk , This library is amazing! I've implemented it in a WinForms project, but I'm encountering an issue where the application exits itself after calling hook.Dispose(). Any thoughts on why this might be happening?

**Reply by TolikPylypchuk** [reply]
Hello! Thanks for asking! Could you please create an issue and provide a minimal reproducible example? A small WinForms application would be ideal. I will then look into it.

**Reply by gulshan-verma** [reply]
Hi @TolikPylypchuk , The problem has been resolved after employing the `hook.RunAsync()` Nonetheless, thank you for responding.

**Comment by Butcher-vj** [comment]
sharpHook.Native Event simulator :
In windows, while trying to simulate a ArrowUp, ArrowLeft, ArrowRight and ArrowDown keys with numlock being on in the remote machine, results in simulating like 8 for ArrowUp and 6 for ArrowRight and 4 for ArrowLeft and 2 for ArrowDown.

**Reply by TolikPylypchuk** [reply]
Let's discuss this in the issue you've created.

**Comment by Butcher-vj** [comment]
Hi @TolikPylypchuk , I am trying to simulate text entry in a machine using using SharpHook.Native.UioHookResult.SimulateTextEntry.
If the string i want to simulate is like this means "112233", the simulatetextEntry is simualting only 123 ?
This is happening everywhere, but if the text is being simulated in a notepad means it works fine but if i use this simulation in like file explorer or browser for searching purpose like that it simulates only non repeating characters.
Is there anyone who faced this kind of issues before ?

**Reply by TolikPylypchuk** [reply]
Hi! Thanks for asking! This looks like a bug. Please create an issue, and I will look into it.
